src/util.ts

```diff
@@ -63,8 +63,8 @@ export function globify(inputFilePath: string | string[]): string | string[] {
   throw new Error('Only file path or directory path are supported.');
 }
 
-export function isBinary(filePath: string, newFileContents?: string | Buffer) {
-  const extension = path.extname(filePath).replace(/^\./, '') || path.basename(filePath);
+export function isBinary(filePath: string, newFileContents?: Buffer) {
+  const extension = path.extname(filePath).replace(/^\./, '');
```

Comment:
Removing the `basename` will break the `textextensions` check with files like `Dockerfile`/`Makefile` and the such.